core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt

```diff
@@ -389,8 +442,8 @@ class MapGenerator(val ruleset: Ruleset) {
     private fun spawnIce(tileMap: TileMap) {
         tileMap.setTransients(ruleset)
         val temperatureSeed = randomness.RNG.nextInt().toDouble()
-        for (tile in tileMap.values) {
-            if (tile.baseTerrain !in Constants.sea || tile.terrainFeatures.any())
+        for (tile in tileMap.values.asSequence()) {
```

Comment:
Wouldn't adding a .asSequence() here actually add an extra layer of abstraction without any gain? So this would be a net loss, no?

Reply:
I vaguely remember evaluating what kind of iterator will result and whether it's the best possible one - for most loops in that file - but here sleep deprivation must have bitten. `values` is not the View of a HashMap but our own, effectively just erasing type information on `private var tileList`, so whatever iterator `Collection` chooses can't be improved on.

So - net loss - maybe, without any gain - yup!